src/main/kotlin/uk/gov/justice/digital/hmpps/prisonregister/model/PrisonFilter.kt

```diff
@@ -21,13 +22,19 @@ class PrisonFilter(
     genders?.forEach {
       andBuilder.add(cb.equal(root.get<Any>(it.columnName), true))
     }
+    if (!prisonTypeCodes.isNullOrEmpty()) {
+      val prisonTypesPredicate =
+        root.join<Any, Any>("prisonTypes").get<Any>("type").`in`(prisonTypeCodes.map { it })
+      andBuilder.add(prisonTypesPredicate)
+    }
     if (!textSearch.isNullOrBlank()) {
       val orBuilder = ImmutableList.builder<Predicate>()
       orBuilder.add(textSearchWildcardAndIgnoreCasePredicate(root, cb, "prisonId"))
       orBuilder.add(textSearchWildcardAndIgnoreCasePredicate(root, cb, "name"))
       andBuilder.add(cb.or(*orBuilder.build().toTypedArray()))
     }
     query.orderBy(cb.asc(root.get<Any>("prisonId")))
+    query.distinct(true)
```

Comment:
The join with `prison_type` meant the query was pulling back duplicate records so included `distinct`

Reply:
It is probably worth adding a test for this scenario (to ensure no duplications)